x/gamm/keeper/pool.go

```diff
@@ -68,8 +68,10 @@ func (k Keeper) GetPoolAndPoke(ctx sdk.Context, poolId uint64) (types.CFMMPoolI,
 	return pool, nil
 }
 
-// Get pool and check if the pool is active, i.e. allowed to be swapped against.
-func (k Keeper) getPoolForSwap(ctx sdk.Context, poolId uint64) (types.CFMMPoolI, error) {
+// GetCFMMPool gets CFMMPool and check if the pool is active, i.e. allowed to be swapped against.
+// The difference from GetPoolIs is that this function returns an error if the pool is inactive
+// and the pool is of CFMMPool type.
```

Comment:
```suggestion
// GetCFMMPool gets CFMMPool and checks if the pool is active, i.e. allowed to be swapped against.
// The difference from GetPool is that this function returns an error if the pool is inactive
// or if the pool is not of type CFMMPool.
```

Reply:
Aside from "check" -> "checks", the rest does not look right to me. I will refactor this to be clearer

Reply:
Done